pkgs/development/python-modules/llmx/default.nix

```
@@ -0,0 +1,71 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  setuptools-scm,
+  setuptools,
+  pythonOlder,
+  pydantic,
+  openai,
+  tiktoken,
+  diskcache,
+  cohere,
+  google-auth,
+  typer,
+  pyyaml,
+  transformers,
+  fastapi,
+  uvicorn,
+  accelerate,
+  withWeb ? true,
+  withTransformers ? true,
+}:
+
+buildPythonPackage rec {
+  pname = "llmx";
+  version = "0.0.21a0";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-OEo6wIaDTktzAsP0rOmhxjFSHygTR/EpcRI6AXsu+6M=";
+  };
+
+  disabled = pythonOlder "3.9";
+
+  build-system = [ setuptools-scm ];
+
+  dependencies =
+    [
+      pydantic
+      openai
+      tiktoken
+      diskcache
+      cohere
+      google-auth
+      typer
+      pyyaml
+      setuptools
+    ]
+    ++ lib.optionals withWeb [
+      fastapi
+      uvicorn
+    ]
+    ++ lib.optionals withTransformers [
+      transformers
+      accelerate
+    ];
```

Remove `withWeb` and `withTransformers` and use `optional-dependencies` please. With `optional-dependencies` will consumers be able to pull what they need `++ llmx.optional-dependencies.web` or `++ llmx.optional-dependencies.transformers`.

```suggestion
    ];

  passthru.optional-dependencies = {
    web = [
       fastapi
       uvicorn
    ];
    transformers = [
      accelerate
      transformers
    ] ++ transformers.optional-dependencies.torch;
  };
```